src/compiler/diagnosticMessages.json

```diff
@@ -4280,6 +4280,10 @@
         "category": "Message",
         "code": 6228
     },
+    "Function-like tag expects more arguments than the JSX factory can provide.": {

```

Comment:
```suggestion
    "'{0}' expects at least '{1}' arguments, but the JSX factory '{2}' provides at most '{3}'.": {
```

Reply:
`at least` isn't quite right for the current implementation - at least right now, we're never looking at minimum argument count here. For example, I could write
```ts
declare function MyTagWithOptionalNonJSXBits(props: Props, context: any, nonReactArg?: string): JSX.Element;
```
and what's here right now would issue an error on that tag's usage, and I very much question if it's OK to feed that into `React`... Minimally, we know `React` will _never_ provide that third argument...

Reply:
OK, I've adjusted the logic to match your message, even if i think it allows some questionable tags through.